www/public-interface.js

```diff
@@ -180,7 +180,14 @@ module.exports = function init(exec, cookieHandler, urlUtil, helpers, globalConf
         break;
       case 'upload':
         var fileOptions = helpers.checkUploadFileOptions(options.filePath, options.name);
-        exec(onSuccess, onFail, 'CordovaHttpPlugin', 'uploadFiles', [url, headers, fileOptions.filePaths, fileOptions.names, options.connectTimeout, options.readTimeout, options.followRedirect, options.responseType, reqId]);
+        var hasProgressCallback = options.onProgress != null;
+        exec(function(resp) {
+          if (resp != null && resp.isProgress) {
+            options.onProgress(resp);
+          } else {
+            onSuccess(resp);
+          }
```

Comment:
Please extract this inline handler into a function expression in lines 183ff for making it more readable.